qemu/tests/hugepage_reset.py

```diff
@@ -21,6 +21,27 @@ def run(test, params, env):
     :param env: Dictionary with the test environment.
     """
 
+    def allocate_largepages_per_node():
+        """
+        The functions is inteded to set 1G hugepages per NUMA
+        node when the system has four or more of these nodes.
+        For this function to work the hugepage size should be 1G.
+        This way a QEMU failure could be avoided if it's unable
+        to allocate memory.
+        """
+        node_list = host_numa_node.online_nodes_withcpumem
+        if len(node_list) >= 4:
+            try:
+                for node in node_list:
+                    node_mem_free = int(
+                        host_numa_node.read_from_node_meminfo(node, "MemFree")
+                    )
+                    mem_kb = mem * 1024
+                    if node_mem_free > mem_kb:
+                        hp_config.set_node_num_huge_pages(4, node, "1048576")
+            except ValueError as e:
+                test.cancel(e)
```

Comment:
I am not sure when we will meet `ValueError`, `read_from_node_meminfo`?
When all nodes do not match the enough memory we want, the current loop will continue the test, but we should skip the test as well.

Reply:
@PaulYuuu 
`ValueError` is hit when `set_node_num_huge_pages` fails allocating hugepages in the node, for example, because there's not enough memory. Added a break and a condition to really check if the test could be run